connector_dns/connector.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Elico Corp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from openerp import models, fields
+from openerp.addons.connector.connector import (Environment)
+from openerp.addons.connector.checkpoint import checkpoint
+
+
+def get_environment(session, model_name, backend_id):
+    """ Create an environment to work with.  """
+    backend_record = session.browse('dns.backend', backend_id)
+    env = Environment(backend_record, session, model_name)
+    return env
+
+
+class DNSBinding(models.AbstractModel):
+    """ Abstract Model for the Bindigs.
+    All the models used as bindings between dnspod and OpenERP
+    (``dnspod.res.partner``, ``dnspod.product.product``, ...) should
+    ``_inherit`` it.
+    """
+    _name = 'dns.binding'
+    _inherit = 'external.binding'
+    _description = 'dns Binding (abstract)'
+
+    dns_backend_id = fields.Many2one(
+        comodel_name='dns.backend',
+        String='DNS Backend',
+        store=True,
+        ondelete='restrict'
+    )
+    # fields.char because 0 is a valid dnspod ID
+    dns_id = fields.Char('ID on other software')
+    # state of the record synchronization with dnspod
+    state = fields.Selection(
```

Comment:
How important is this column to your internal business logic? Reason I ask is that I am finding that it is somewhat ambiguous in the following examples, where using the `sync_date` as the status would eliminate the ambiguity:

* Record is synced to remote system. It is then changed on the remote system. Record is out of sync w/ connector, while being indicated that sync is done
* Record is synced to remote system. It is then changed in Odoo, resynced, and failed. There is now no indication that the record originally passed sync

I do think I like the idea of knowing about failures though. What if we added:

* `fail_date` - this column is the inverse of `sync_date`, providing the last sync fail time